src/qibolab/instruments/qm/controller.py

```diff
@@ -373,6 +375,10 @@ def sweep(self, qubits, couplers, sequence, options, *sweepers):
         # play pulses using QUA
         with qua.program() as experiment:
             n = declare(int)
+            for qubit in qubits.values():
+                if qubit.flux:
+                    qua.set_dc_offset(qubit.flux.name, "single", qubit.sweetspot)
```

Comment:
In this case it is not really redundant, as without it we are not setting the proper offset. I am not sure if there is any disadvantage, most likely there will be some overhead in the program execution time, however this should be minimal and it happens before starting the actual experiment so it will not affect pulse scheduling.